services/cases.js

```diff
@@ -190,27 +190,36 @@ function getIdCase (query,callback) {
 
 async function getCaseSummary(query, user, callback) {
   try {
-    const caseAuthors = await thisUnitCaseAuthors(user)
-    const scope = Check.countByRole(user, caseAuthors)
-    const filter = await Filter.filterCase(user, query)
-    const searching = Object.assign(scope, filter)
-    const { sumFuncNoMatch } = require('../helpers/aggregate/func')
-    const conditions = [
-      { $match: {
-        $and: [  searching, { ...WHERE_GLOBAL, last_history: { $exists: true, $ne: null } } ]
-      }},
-      {
-        $group: {
-          _id: 'status',
-          confirmed: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.CONF] }]),
-          probable: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.PROB] }]),
-          suspect: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.SUS] }]),
-          closeContact: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.CLOSE] }]),
-        },
-      },{ $project: { _id : 0 } },
-    ]
-    const result = await Case.aggregate(conditions)
-    callback(null, result.shift())
+    clientConfig.get(`summary-cases-list-${user.username}`, async (err, result) => {
+      const caseAuthors = await thisUnitCaseAuthors(user)
+      const scope = Check.countByRole(user, caseAuthors)
+      const filter = await Filter.filterCase(user, query)
+      const searching = Object.assign(scope, filter)
+      const { sumFuncNoMatch } = require('../helpers/aggregate/func')
+      if(result){
+        const resultJSON = JSON.parse(result)
+        callback(null, resultJSON)
+      }else{
+        const conditions = [
+          { $match: {
+            $and: [  searching, { ...WHERE_GLOBAL, last_history: { $exists: true, $ne: null } } ]
+          }},
+          {
+            $group: {
+              _id: 'status',
+              confirmed: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.CONF] }]),
+              probable: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.PROB] }]),
+              suspect: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.SUS] }]),
+              closeContact: sumFuncNoMatch([{ $eq: ['$status', CRITERIA.CLOSE] }]),
+            },
+          },{ $project: { _id : 0 } },
+        ]
+        const result = await Case.aggregate(conditions)
+        const shiftResult = result.shift()
+        clientConfig.setex(`summary-cases-list-${user.username}`, 15 * 60 * 1000, JSON.stringify(shiftResult)) // set redis key
```

Comment:
```suggestion
        clientConfig.setex(`summary-cases-list-${user.username}`, 15 * 60 * 1000, JSON.stringify(shiftResult)) // set redis key
```
Mangstaapp @firmanJS @sandisunandar99 👍

Minor catatan aja bagian expiry time itu dibuat jadi variable aja, dikasih nama/comment yang jelas kalau itu maksudnya adalah durasi cache sebelum expired 👍😁

Reply:
yoi mas,, untuk sementara supaya maredakan masa,, next refactor..
tinggal 1 lagi mas bkin redaction last update data dashboar supaya user lebih awkarin karena datanya cache,, :dancers: :+1: :laughing: